source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/SqlStatements/Mappers/SettlementMethodMapper.cs

```diff
@@ -36,4 +36,13 @@ public static class SettlementMethodMapper
             TimeSeriesType.TempProduction => null,
             _ => throw new NotImplementedException($"Cannot map timeSeriesType type '{timeSeriesType}"),
         };
+
+    public static SettlementMethod? FromDeltaTableValue(string settlementMethod) =>
+        settlementMethod switch
+        {
+            "flex" => SettlementMethod.Flex,
+            "non_profiled" => SettlementMethod.NonProfiled,
+            "" => null,
+            _ => throw new NotImplementedException($"Cannot map settlementMethod type '{settlementMethod}"),
```

Comment:
If we plan to surround the type with ' then we should also consider this "fix" for the "timeSeriesType" above.

```suggestion
            _ => throw new NotImplementedException($"Cannot map settlementMethod type '{settlementMethod}'"),
```